common/src/lib.rs

```diff
@@ -2,15 +2,79 @@ pub mod crypto;
 
 use crypto::*;
 use serde::{Deserialize, Serialize};
-use std::collections::BTreeSet;
+use std::collections::{BTreeMap, BTreeSet};
 use thiserror::Error;
 
+/// The state that is directly recorded in the header.
+///
+/// This state affects the consensus, unlike the ordinary state (which is used for data recording).
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct EssentialState {
+    /// The original validator set information before the delegation calculation.
+    ///
+    /// The order here is the leader selection priority order of the validators
+    /// which directly affects the result of `calculate_net_validator_set()`.
+    pub validator_set: Vec<(PublicKey, u64)>,
```

Comment:
What do you think about aliasing `VotingPower = u64`? It will people understand the code at a glance.

Reply:
fixed.